### PR TITLE
[JENKINS-62231] Move start logic to postInitialize to avoid OldDataMo…

### PR DESCRIPTION
**Pull request by MRamonLeon**
See [JENKINS-62231](https://issues.jenkins-ci.org/browse/JENKINS-62231) for full story

To avoid the plugin to fail when loading its configuration if any modification has been done we move the logic from `start` to `postInitialize` method. It may help for future versions upgrade/downgrade ability.

@res0nance @fcojfernandez @alecharp @varyvol @rsandell ARggghhhh I cannot even request reviewers

### CODE REVIEW

File: src/main/java/hudson/plugins/ec2/PluginImpl.java [hunk 1]
```diff
@@ -85,7 +76,15 @@ public String getDisplayName() {
...
     @Override
-    public void postInitialize() {
+    public void postInitialize() throws IOException {
+        // backward compatibility with the legacy class name
+        Jenkins.XSTREAM.alias("hudson.plugins.ec2.EC2Cloud", AmazonEC2Cloud.class);
+        Jenkins.XSTREAM.alias("hudson.plugins.ec2.EC2Slave", EC2OndemandSlave.class);
+        // backward compatibility with the legacy instance type
+        Jenkins.XSTREAM.registerConverter(new InstanceTypeConverter());
+
+        load();
```
Comment (MRamonLeon) [inline src/main/java/hudson/plugins/ec2/PluginImpl.java:86]: maybe just `load()` is needed

**Comment by MRamonLeon** [comment]
The build is failing because the [recommendedConfiguration uses jenkins.version=2.164.1](https://github.com/jenkins-infra/pipeline-library/blob/master/vars/buildPlugin.groovy#L258) and [we updated jenkins.version to 2.176.4](https://github.com/jenkinsci/ec2-plugin/pull/454/files#diff-600376dffeb79835ede4a0b285078036R86). We should use a different configuration or revert the bump of the jenkins.version

**Comment by MRamonLeon** [comment]
I believe my changes on Jenkinsfile wasn't taken into account

**Comment by res0nance** [comment]
There seems to be an issue on windows as well, I'll work on it later, right now I'm trying to get the builds going again quickly

**Comment by MRamonLeon** [comment]
@res0nance green again, great!! Do you think you would be able to cut a release with this one and more interesting, with #455, to avoid pain on people upgrading with old ssh clients?

**Comment by MRamonLeon** [comment]
@res0nance would you be able to cut a release with #455 and other improvements? To avoid pain on people upgrading with old ssh clients.

**Pull request merged**